network/src/peers/peers.rs

```diff
@@ -142,42 +142,33 @@ impl<S: Storage + Send + Sync + 'static> Node<S> {
     }
 
     ///
-    /// Broadcasts a connection request to all default bootnodes of the network.
+    /// Broadcasts a connection request to all the supplied addresses.
     ///
-    /// This function attempts to reconnect this node server with any bootnode peer
-    /// that this node may have failed to connect to.
-    ///
-    /// This function filters out any bootnode peers the node server is
+    /// This function filters out any peers the node server is
     /// either connnecting to or already connected to.
     ///
-    async fn connect_to_bootnodes(&self) {
+    pub async fn connect_to_addresses(&self, addrs: &[SocketAddr]) {
         // Local address must be known by now.
         let own_address = self.local_address().unwrap();
 
-        // Iterate through each bootnode address and attempt a connection request.
-        for bootnode_address in self
-            .config
-            .bootnodes()
-            .iter()
-            .filter(|peer| **peer != own_address)
-            .copied()
-        {
-            let node = self.clone();
-            if node.peer_book.is_connected(bootnode_address) {
-                return;
+        for node_addr in addrs.iter().filter(|addr| **addr != own_address).copied() {
+            if self.peer_book.is_connected(node_addr) {
+                break;
```

Comment:
this PR doesn't change this, but I just realized that this should be a part of the `filter` predicate as opposed to a `break`, no? we don't want to disregard the rest of the list as soon as we encounter an active connection after all

Reply:
This used to be a `return`. I agree though, this should be checked during filtering.